addons/webhooks/clusterbootstrap_webhook_test.go

```diff
@@ -0,0 +1,200 @@
+			Expect(clusterBootstrap.Spec.CSI.RefName).To(Equal(clusterBootstrapTemplate.Spec.CSI.RefName))
+		})
+		It("should add defaults ONLY to the missing fields when the ClusterBootstrap CR has the predefined annotation", func() {
+			// Create a ClusterBootstrap with empty spec
+			clusterBootstrap := &runv1alpha3.ClusterBootstrap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clusterBootstrapName,
+					Namespace: clusterBootstrapNamespace,
+					Annotations: map[string]string{
+						constants.AddCBMissingFieldsAnnotationKey: "v1.23.5---vmware.1-tkg.1-zshippable",
+					},
+				},
+				Spec: &runv1alpha3.ClusterBootstrapTemplateSpec{
+					CSI: &runv1alpha3.ClusterBootstrapPackage{
+						RefName: fmt.Sprintf("%s.%s", fakeCarvelPackageRefName, fakeCarvelPackageVersion),
+						ValuesFrom: &runv1alpha3.ValuesFrom{
+							Inline: map[string]interface{}{
+								"foo": "bar",
+							},
+						},
+					},
+				},
+			}
+			err := k8sClient.Create(ctx, clusterBootstrap)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = k8sClient.Get(ctx, client.ObjectKey{Namespace: clusterBootstrapNamespace, Name: clusterBootstrapName}, clusterBootstrap)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(clusterBootstrap.Spec).NotTo(BeNil())
+			Expect(clusterBootstrap.Spec.CNI.RefName).To(Equal(clusterBootstrapTemplate.Spec.CNI.RefName))
+			Expect(clusterBootstrap.Spec.Kapp.RefName).To(Equal(clusterBootstrapTemplate.Spec.Kapp.RefName))
+			Expect(clusterBootstrap.Spec.CSI.RefName).To(Equal(clusterBootstrapTemplate.Spec.CSI.RefName))
+			Expect(clusterBootstrap.Spec.CSI.ValuesFrom.Inline["foo"]).To(Equal("bar"))
+		})
+		It("should NOT add defaults to the missing fields when the ClusterBootstrap CR does not have the predefined annotation", func() {
+			// Create a ClusterBootstrap with empty spec
+			clusterBootstrap := &runv1alpha3.ClusterBootstrap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clusterBootstrapName,
+					Namespace: clusterBootstrapNamespace,
+				},
+				Spec: &runv1alpha3.ClusterBootstrapTemplateSpec{},
+			}
+			err := k8sClient.Create(ctx, clusterBootstrap)
+			Expect(err).To(HaveOccurred())
+			Expect(apierrors.IsInvalid(err)).To(BeTrue())
+		})
+		It("should NOT add defaults to the missing fields when the predefined annotation has invalid value", func() {
+			// Create a ClusterBootstrap with empty spec
+			clusterBootstrap := &runv1alpha3.ClusterBootstrap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clusterBootstrapName,
+					Namespace: clusterBootstrapNamespace,
+					Annotations: map[string]string{
+						constants.AddCBMissingFieldsAnnotationKey: "fake-tkr-does-not-exist",
+					},
+				},
+				Spec: &runv1alpha3.ClusterBootstrapTemplateSpec{},
+			}
+			err := k8sClient.Create(ctx, clusterBootstrap)
+			Expect(err).To(HaveOccurred())
+			Expect(apierrors.IsNotFound(err)).To(BeTrue())
+		})
+
+		// TODO: Add more tests to verify the CompleteCBPackageRefNamesFromTKR() logic.
```

Comment:
Could you file an issue?

Reply:
Done. I made some changes to TKR CRD and fixed the issue.